### PR TITLE
LuxoPremier Api based webApp

### PR DESCRIPTION
**Pull request by Progress2002**
In this Project, we collaborated together to create LuxoPremier, A web app based on tv-shows using Tv-maze and Involvement API
we have followed these [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md)

here is the link to our presentation of the project [video](https://drive.google.com/file/d/1H5PkA-i1_EXW1Vu3j3ECRZrKJuYUP3YE/view?usp=sharing)

### Type of change

-  Bug fix (non-breaking change which fixes an issue)
-  Create a show container and dynamically populate the section with data gotten from an external API
-  Added likes functionality that enables ures to like a movie
- Created a function that counts the number of likes and displays them in the likes counter
- Style show cards
-  Create projects modal
-  Style projects modal
- Added a comment form section that enables users to add a new comment to a selected movie
- Created a function that counts the number of comments and displays them in the comment counter
- Added unit testing for comments and likes counter.

### Checklist:

-  Our code follows the style guidelines of this project
-  We have performed a self-review of our code
-  We have made corresponding changes to the documentation
-  Our changes generate no new warnings
-  We have added tests that prove our fix is effective or that our feature works
-  New and existing unit tests pass locally with our changes
-  Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/modules/counter.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+ * @jest-environment jsdom
+ */
+import { commentsCount, showsCount, likesCount } from '../modules/counter'  //eslint-disable-line
```
Comment (Ghiftee) [inline src/modules/counter.test.js]: - [ ] I suggest you do not disable linters as they help us write cleaner code.

File: src/modules/counter.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    const shows = ['show1', 'show2', 'show3', 'show4'];
+    const result = showsCount(shows);
+    expect(result).toBe(4);
+  });
+
+  test('Assume previous likes is "14" expect "15" ', () => {
+    document.body.innerHTML = '<span class="like-symbol">14</span>';
+    expect(likesCount(14)).toBe(15);
+  });
+});
```
Comment (Ghiftee) [inline src/modules/counter.test.js]: - [ ] Kindly add more tests to cover all edge cases. For instance, a scenario in which there are 0 items on the page, in this case, counter function should return 0.

File: src/modules/homepage.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import CARD_API from './apiUrl.js';
+import { postLike, getLikesData } from './likeApi.js';
+import selectedCardDetails from './modal.js'; //eslint-disable-line
```
Comment (Ghiftee) [inline src/modules/homepage.js]: - [ ]  I suggest you do not disable linters as they help us write cleaner code.

File: src/modules/modal.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import CARD_API from './apiUrl.js';
+import { getApiData } from './homepage.js'; //eslint-disable-line
```
Comment (Ghiftee) [inline src/modules/modal.js]: - [ ]  I suggest you do not disable linters as they help us write cleaner code.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello team 👋🏽 

Good job working on the project so far! 
✔️ Your app works as expected
✔️ Your project is documented professionally
✔️ Your tests pass
However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by mwanawabangona — APPROVED** [review]
Hi @Team!

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job Implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
